darshan-util/pydarshan/tests/test_summary.py

```diff
@@ -110,6 +110,17 @@ def test_main_without_args(tmpdir, argv):
                         # check that help message is present
                         assert "Heat map is not available for this job" in report_str
 
+                    # check that expected number of figures are found
+                    actual_img_count = 0
+                    for txt in report_str.split():
+                        if "img" in txt:
+                            actual_img_count += 1
+                    assert actual_img_count == expected_img_count
```

Comment:
I think this is a fair bit clearer/faster/more idiomatic:

```diff
--- a/darshan-util/pydarshan/tests/test_summary.py
+++ b/darshan-util/pydarshan/tests/test_summary.py
@@ -111,11 +111,7 @@ def test_main_without_args(tmpdir, argv, expected_img_count):
                         assert "Heat map is not available for this job" in report_str
 
                     # check that expected number of figures are found
-                    actual_img_count = 0
-                    for txt in report_str.split():
-                        if "img" in txt:
-                            actual_img_count += 1
-                    assert actual_img_count == expected_img_count
+                    assert report_str.count("img") == expected_img_count

```

Apart from that, the bug in the plotting code has a separate issue and I haven't seen any objections from Argonne folks re: the basic layout now, so close to ready I think.

Reply:
Sorry, just now getting caught up on this and #581, but think the end result looks good. No objections here on merging once the bug I reported is fixed.

Reply:
I've add the suggested change here.